### PR TITLE
feat: add burn-in to test execution logic

### PR DESCRIPTION
**Pull request by MuazOthman**
- Closes CYCLOUD-1665

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/driver/src/cypress/mocha.ts [hunk 1]
```diff
@@ -73,10 +117,17 @@ export function calculateTestStatus (test: CypressTest, strategy: Strategy, opti
     failedTests.push(test)
   }
 
+  if (!test.prevAttempts?.length) {
+    test.thisAttemptInitialStrategy = 'NONE'
```
Comment (cacieprins) [inline packages/driver/src/cypress/mocha.ts:120]: `thisAttemptInitialStrategy` is never referenced, from what I can tell - is this a field we need to be reporting to the cloud?

File: packages/driver/src/cypress/mocha.ts [hunk 2]
```diff
@@ -73,10 +117,17 @@ export function calculateTestStatus (test: CypressTest, strategy: Strategy, opti
...
+  } else {
+    test.thisAttemptInitialStrategy = test.prevAttempts[test.prevAttempts.length - 1].nextAttemptStrategy
+  }
+
   // If there is AT LEAST one failed test attempt, we know we need to apply retry logic.
   // Otherwise, the test might be burning in (not implemented yet) OR the test passed on the first attempt,
   // meaning retry logic does NOT need to be applied.
   if (failedTests.length > 0) {
+    test.nextAttemptStrategy = 'RETRY'
```
Comment (cacieprins) [inline packages/driver/src/cypress/mocha.ts:129]: `nextAttemptStrategy` is only referenced to set `thisAttemptStrategy`; is this logic necessary?

File: packages/driver/src/cypress/mocha.ts [hunk 3]
```diff
@@ -95,40 +146,68 @@ export function calculateTestStatus (test: CypressTest, strategy: Strategy, opti
...
-      )) {
-      test.final = false
+    switch (strategy) {
+      case 'detect-flake-and-pass-on-threshold':
+        if (passingAttempts >= (passesRequired as number)) {
+          // we met the threshold, so we can stop retrying and pass the test
+          outerTestStatus = 'passed'
+          test.final = true
+          shouldAttemptsContinue = false
+          test.reasonToStop = 'PASSED_MET_THRESHOLD'
```
Comment (cacieprins) [inline packages/driver/src/cypress/mocha.ts:155]: `reasonToStop` is only set, and never referenced - is this a field we'll be sending to the cloud?

File: packages/driver/src/cypress/mocha.ts (hunk 1 above)
Comment (MuazOthman) [reply packages/driver/src/cypress/mocha.ts:120]: Yes, see the description [here](https://cypress-io.atlassian.net/browse/CYCLOUD-1665)

File: packages/driver/src/cypress/mocha.ts (hunk 2 above)
Comment (MuazOthman) [reply packages/driver/src/cypress/mocha.ts:129]: This is so we can indicate if the next attempt is to be made as part of a retry. I'm not sure if there's another way to do it.

File: packages/driver/src/cypress/mocha.ts (hunk 3 above)
Comment (MuazOthman) [reply packages/driver/src/cypress/mocha.ts:155]: Yes, see the description [here](https://cypress-io.atlassian.net/browse/CYCLOUD-1665)

**Pull request closed without merging**